InformationScripting/src/queries/CompositeQuery.h

```diff
@@ -66,8 +73,7 @@ class INFORMATIONSCRIPTING_API CompositeQuery : public Query
 		};
 
 		struct QueryNode {
-				QueryNode(Query* q) : q_{q} {}
-				~QueryNode();
+				QueryNode(std::unique_ptr<Query> q) : q_{std::move(q)} {}
```

Comment:
Better just get the argument as `&&` and not use move. This expresses the intent more clearly. You might need in this case to use `std::forward` in the client code: http://en.cppreference.com/w/cpp/utility/forward